### DISCUSSION TITLE
How to integrate a PNG image as part of CreateSeuratObject

### DISCUSSION THREAD

**Discussion by derek-atlas** [satijalab/seurat#4231] [Q&A]
I am following along with the "Analysis, visualization, and integration of spatial datasets with Seurat" vignette (https://satijalab.org/seurat/articles/spatial_vignette.html) with my own gene expression matrices. I am trying to create the Spatial DimPlot where the clusters are overlaid on top of my microscope tissue slide image but am given the following error when running 
```p1 <- DimPlot(brain, reduction = "umap", label = TRUE)
p2 <- SpatialDimPlot(brain, label = TRUE, label.size = 3)
p1 + p2
```

```
Error in SpatialPlot(object = object, group.by = group.by, images = images,  : 
  Could not find any spatial image `information`
```
  
The error makes sense as I have not included an image when using the CreateSeuratObject. Rather, I've been using ggplot2 and the OpenImageR package to add the background images. How should I go about incorporating the image into the Seurat object like what was done with the 10x practice dataset? The code I'm using for creating my object is shown below. 

```
seuratobj <- CreateSeuratObject(gem.transposed, min.cells = min.tixels)
  
  seuratobj <- PercentageFeatureSet(seuratobj,
                                    pattern =  mt.pattern, 
                                    col.name = "percent.mt")
```

**Comment by timoast** [comment]
Hi @derek-atlas, we're currently thinking about how best to support custom spatial assays in Seurat. In re-designing our spatial data support, it would be useful to know some common data types that users have. Could you briefly explain what type of data you have, and how the transcriptional measurements relate to regions in the image?

**Reply by bayesbm** [reply]
HI @timoast , Today I came across this thread, as I run into similar problems. Have you made any progress with this? As I am manually creating a Seurat object using 'CreateSeuratObject', it would be valuable to have a resource. All the best, Bas

**Comment by TangentDeng** [comment]
@BasMatsuura hi, does this issue solved?  I had my own merfish data and tiff image. How to create a custom seurat file with spatial image? I only can make the spatial postion display as a DimReducObject like this: 
test[["spatial"]] <- CreateDimReducObject(embeddings = as.matrix(posi), 
                                                key = "Spatial",assay = "RNA")
DimPlot(test,reduction = "spatial",pt.size = 1)
but when:
SpatialFeaturePlot(test, 
                   features = "nCount_RNA")
Error in SpatialPlot(object = object, features = features, images = images,  : 
  Could not find any spatial image information
